doc/api/animation_api.rst

```diff
@@ -124,7 +125,36 @@ artist at a global scope and let Python sort things out.  For example ::
    plt.show()
 
 The second method is to use `functools.partial` to 'bind' artists to
-function.  A third method is to use closures to build up the required
+function. ::
+
+   import numpy as np
+   import matplotlib.pyplot as plt
+   from matplotlib.animation import FuncAnimation
+   from functools import partial
+
+   fig, ax = plt.subplots()
+   xdata, ydata = [], []
+   ln, = plt.plot([], [], 'ro')
+
+   def init():
+       ax.set_xlim(0, 2*np.pi)
+       ax.set_ylim(-1, 1)
+       return ln,
+
+   def update(frame):
+       xdata.append(frame)
+       ydata.append(np.sin(frame))
+       ln.set_data(xdata, ydata)
+       return ln,
+
+   ani = FuncAnimation(
+    fig, partial(update, offset=-0.5),
```

Comment:
I guess I'm a good guinea pig for this, as I don't really know what `partial` is supposed to do.  After this I am still mystified :wink:.  Why wouldn't I just pass `update` directly here?  I don't see where `offset` ever gets used, so I don't see what the -0.5 means.

Reply:
Ah, my bad. I forgot to update `update()`, I'll make this clearer.

Reply:
This should address my issue too

Reply:
This example is also using closures (as `update` is closing over `xdata` and `ydata`).  Maybe keep this example dropping the partial to show the closure and switch this example to something like:

```python

def update(frame, x, y):
    x.append(...)
    y.append(....)
    ...

xdata = []
ydata = []
ani = FuncAnimation(fig, partial(update, x=xdata, y=ydata), ...)

```

Reply:
This makes more sense, thanks for the input